src/sentry/static/sentry/app/components/events/interfaces/keyValueList/keyValueList.tsx

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import styled from '@emotion/styled';
+import sortBy from 'lodash/sortBy';
+
+import ContextData from 'app/components/contextData';
+import theme from 'app/utils/theme';
+
+type Props = {
+  /**
+   * Data to render, could be almost anything as the various event interfaces
+   * support all sorts of data.
+   */
+  data: undefined | null | Record<string, any> | any[];
+  /**
+   * Whether or not the data should be rendered with ContextData
+   */
+  isContextData?: boolean;
+  /**
+   * Should the keys be sorted.
+   */
+  isSorted?: boolean;
+  /**
+   * Should the raw text be rendered.
+   */
+  raw?: boolean;
+  /**
+   * Enable wider rendering of the keys.
+   */
+  longKeys?: boolean;
+  onClick?: (event: React.MouseEvent<HTMLElement>) => void;
+};
+
+/**
+ * Generic KeyValue data renderer. The V2 version
+ * of this component can also render annotations
+ * for datascrubbing.
+ */
+function KeyValueList({
+  data,
+  onClick,
+  isContextData = false,
+  isSorted = true,
+  raw = false,
+  longKeys = false,
+}: Props) {
+  // TODO(dcramer): use non-string keys as reserved words ("unauthorized")
+  // break rendering
+
+  if (data === undefined || data === null) {
+    data = [];
```

Comment:
👎 to masking parameters, though looks like it already did this, so neautral.

Reply:
I agree, I'm not excited by it either. I was aiming to change the runtime code as little as possible. This is part of issue details, handles way too many types of data, and I didn't want to chance a regression

Reply:
Yeah understandable